### PR TITLE
[WIP] Add Montreal

### PR DESCRIPTION
**Pull request by dmitriy-serdyuk**
Start adding the city of Montreal. 

TODO:
- [ ] Add on street parking info
- [ ] Fix biking paths
- [ ] Fix stop signs/lights

Related: #457

### CODE REVIEW

**Comment by dmitriy-serdyuk** [comment]
I am having a problem that this style (e.g. tagged `traffic_signals:direction=backward`) of [traffic lights](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dtraffic_signals) is not imported correctly sometimes
image

They are converted into stop signs.

**Comment by dabreegster** [comment]
The traffic signal issue looks like a bug in importing. I found an example at https://www.openstreetmap.org/node/8319766124 -- is the case you're seeing similar? I'll work on fixing this.

https://github.com/dabreegster/abstreet/blob/56101addcf1ce327486cee5ae6a2a553764d04fc/convert_osm/src/extract.rs#L72

File: importer/config/montreal/cfg.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "osm_url": "http://download.geofabrik.de/north-america/canada/quebec-latest.osm.pbf",
+  "map_config": {
+    "driving_side": "Right",
+    "bikes_can_use_bus_lanes": true,
+    "inferred_sidewalks": true,
+    "separate_cycleways": false
+  },
+  "onstreet_parking": {
+    "SomeAdditionalWhereNoData": {
```
Comment (dabreegster) [inline importer/config/montreal/cfg.json:10]: This is fine to start, but it's a tradeoff -- some roads will wind up much wider than they really are, which can exasperate geometry problems. But it's pretty easy to notice and fix -- you can just go explicitly tag street parking there, pull down fresh .osm, and import again.

**Comment by dmitriy-serdyuk** [comment]
Here is another example: https://www.openstreetmap.org/node/3372083583 . Not tagged for direction, but it should be `forward` by default.

**Comment by dmitriy-serdyuk** [comment]
ABstreet underestimates how tightly people able to park on the street :) I counted about twice as many cars parked compared to the visualization.

**Comment by dabreegster** [comment]
> Here is another example

Thanks! Slow start this morning, looking into this now...

> ABstreet underestimates how tightly people able to park on the street

There are hardcoded values right now: https://github.com/dabreegster/abstreet/blob/56101addcf1ce327486cee5ae6a2a553764d04fc/map_model/src/objects/lane.rs#L13

I could make these configurable per-map as a start to improving it, since people probably tend to park more tightly in denser areas where that's a typical need.

**Comment by dabreegster** [comment]
Sync past https://github.com/dabreegster/abstreet/commit/e7e8bb2ca0f6d2fba2cf0ff42b7d7bd407de99b5 and see if that fixes the traffic signals. I didn't test in Montreal, but I think the same problem was also happening in Cambridge.

**Comment by dabreegster** [comment]
Screenshot from 2021-01-25 10-08-20
Cars in abst get a random length between 4.5m and 6.5m (and buses are hardcoded at 12.5m, for the record). The current values of the parking spot lengths allow the longest cars to fit. From the simulation perspective, squeezing long cars in small spots works fine. Visually, they might overlap a bit, but that's fine. I'm adding a new config field for you to override this for greater density.

**Comment by dmitriy-serdyuk** [comment]
> Sync past [e7e8bb2](https://github.com/dabreegster/abstreet/commit/e7e8bb2ca0f6d2fba2cf0ff42b7d7bd407de99b5) and see if that fixes the traffic signals. I didn't test in Montreal, but I think the same problem was also happening in Cambridge.

Thanks, works now.

**Comment by dabreegster** [comment]
https://github.com/dabreegster/abstreet/commit/f69335edf052540ca23d8ccdc177474bb7be6e50 adds a new field to `cfg.json`. You can set `"street_parking_spot_length": 8.0` there to a smaller value to squeeze more parking spots into a lane.

**Comment by matkoniecz** [comment]
> I was checking OSM for Montreal. It is quite low quality: number of lanes, parking, restricted left turns, u-turns, speed limits. It will take some effort to fix OSM before exporting something decent.

\<advertisement>

StreetComplete android app may be useful - it asks about some detail such as lane count, maxspeed (if you anable this), crossing type and so one, useful for surveys while walking - https://github.com/streetcomplete/StreetComplete

\</advertisement>

**Comment by dabreegster** [comment]
Sorry, crazy few days on my end -- are you blocked by me on anything? Should we merge this PR and start importing Montreal now, and keep iterating later?

**Comment by dmitriy-serdyuk** [comment]
I was a bit busy too, sorry.

I rebased and added parking length 6.5m. Should be good for now. Let's merge for now.

**Comment by dabreegster** [comment]
Thanks! Map looks great, I'll add a commit with the uploaded data in a moment.

**Pull request merged**